api/models/itemDao.js

```diff
@@ -0,0 +1,51 @@
+const dataSource = require("./dataSource");
+
+const checkIfItemExistsById = async (itemId) => {
+  const [result] = await dataSource.query(
+    `
+    SELECT EXISTS(
+      SELECT id 
+      FROM items 
+      WHERE id = ?
+    ) AS value
+  `,
+    [itemId]
+  );
+
+  return !!parseInt(result.value);
+};
+
+const getItemDetailsById = async (itemId) => {
+  return await dataSource.query(
+    `
+    SELECT
+      i.author_name AS author_name,
+      i.item_name AS item_name,
+      i.production_year AS production_year,
+      imj.materials AS materials,
+      i.width AS width,
+      i.length AS length,
+      i.height AS height,
+      i.weight AS weight,
+      i.description AS description,
+      i.starting_bid AS starting_bid,
+      i.image_url AS image_url
+    FROM items AS i
+    JOIN (
+      SELECT
+        item_id,
+        JSON_ARRAYAGG(m.name) AS materials
+      FROM items_materials AS im
+      JOIN materials AS m ON m.id = im.material_id
+      GROUP BY item_id
+    ) imj ON imj.item_id = id
+    WHERE id = ?
+  `,
+    [itemId]
+  );
+};
```

Comment:
```suggestion
const getItemDetailsById = async (itemId) => {
  return dataSource.query(
    `
    SELECT
      i.author_name AS author_name,
      i.item_name AS item_name,
      i.production_year AS production_year,
      imj.materials AS materials,
      i.width AS width,
      i.length AS length,
      i.height AS height,
      i.weight AS weight,
      i.description AS description,
      i.starting_bid AS starting_bid,
      i.image_url AS image_url
    FROM items AS i
    JOIN (
      SELECT
        item_id,
        JSON_ARRAYAGG(m.name) AS materials
      FROM items_materials AS im
      JOIN materials AS m ON m.id = im.material_id
      GROUP BY item_id
    ) imj ON imj.item_id = i.id
    WHERE i.id = ?
  `,
    [itemId]
  );
};
```
동작하는데는 지장없겠지만 가독성상 어떤 테이블의 id컬럼인지 명확히 명시해주는것이 좋습니다.